server/handler/ai/db.go

```diff
@@ -0,0 +1,148 @@
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			return nil, errors.New("no reports found")
+		}
+		return nil, err
+	}
+	return &lastReport, nil
+}
+
+func GetReportByWeek(db *gorm.DB, weekStart, weekEnd time.Time) (*models.Report, error) {
+	var report models.Report
+	if err := db.Where("created_at BETWEEN ? AND ?", weekStart, weekEnd).First(&report).Error; err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			return nil, errors.New("no report found for the specified week")
+		}
+		return nil, err
+	}
+	return &report, nil
+}
+
+// GetAllReports fetches all reports from the database, ordered by creation date descending.
+func GetAllReports(db *gorm.DB) ([]models.Report, error) {
+	var reports []models.Report
+	if err := db.Order("created_at desc").Find(&reports).Error; err != nil {
+		return nil, err
+	}
+	return reports, nil
+}
+
+// GenerateReport creates a new report based on the latest 7days PRs and issues data and save it in database.
+func GenerateReport(db *gorm.DB) (models.Report, error) {
+	endTime := time.Now()
+	startTime := endTime.AddDate(0, 0, -7)
+
+	// Fetch merged pull requests
+	var pullRequests []models.PullRequest
+	if err := db.Where("state = ? AND created_at BETWEEN ? AND ?", "MERGED", startTime, endTime).
+		Preload("Author").Find(&pullRequests).Error; err != nil {
+		return models.Report{}, err
+	}
+
+	// Extract required fields from pull requests
+	var formattedPullRequests []map[string]interface{}
+	for _, pr := range pullRequests {
+		var authorData map[string]interface{}
+		if pr.Author != nil {
+			authorData = map[string]interface{}{
+				"login": pr.Author.Login,
+				"name":  pr.Author.Name,
+				"bio":   pr.Author.Bio,
+			}
+		}
+		formattedPullRequests = append(formattedPullRequests, map[string]interface{}{
+			"updatedAt":    pr.UpdatedAt,
+			"repositoryID": pr.RepositoryID,
+			"title":        pr.Title,
+			"author":       authorData,
+		})
+	}
+
+	// Fetch open issues
+	var issues []models.Issue
+	if err := db.Where("state = ? AND created_at BETWEEN ? AND ?", "OPEN", startTime, endTime).
+		Preload("Author").Find(&issues).Error; err != nil {
+		return models.Report{}, err
+	}
+
+	// Extract required fields from issues
+	var formattedIssues []map[string]interface{}
+	for _, issue := range issues {
+		var authorData map[string]interface{}
+		if issue.Author != nil {
+			authorData = map[string]interface{}{
+				"login": issue.Author.Login,
+				"name":  issue.Author.Name,
+				"bio":   issue.Author.Bio,
+			}
+		}
+		formattedIssues = append(formattedIssues, map[string]interface{}{
+			"updatedAt":    issue.UpdatedAt,
+			"repositoryID": issue.RepositoryID,
+			"title":        issue.Title,
+			"author":       authorData,
+		})
+	}
+
+	// Prepare response data
+	responseData := map[string]interface{}{
+		"pullRequests": formattedPullRequests,
+		"issues":       formattedIssues,
+	}
+
+	// fmt.Printf("Choices: %+v\n", responseData)
+
+	responseJSON, err := json.Marshal(responseData)
```

Comment:
I think you should rename this variable is not really a reponseJSON, more like an input ?

Reply:
same for responseData